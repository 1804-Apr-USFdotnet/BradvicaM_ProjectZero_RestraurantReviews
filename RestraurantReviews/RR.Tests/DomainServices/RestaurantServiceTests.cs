using ApprovalTests;
using ApprovalTests.Reporters;
using Autofac;
using Microsoft.VisualStudio.TestTools.UnitTesting;
using Moq;
using RR.Console;
using RR.DomainContracts;
using RR.DomainServices;
using RR.Models;
using RR.RepositoryContracts;

namespace RR.Tests.DomainServices
{
    [TestClass]
    public class RestaurantServiceTests
    {
        private readonly Mock<IRestaurantRepository> _mock;

        public RestaurantServiceTests()
        {
            _mock = new Mock<IRestaurantRepository>();
            _mock.Setup(x => x.Add(It.IsAny<Restaurant>()));
        }

        [TestMethod]
        [UseReporter(typeof(DiffReporter))]
        public void TopThreeRestaurantByAverageReview_Returns_CorrectRestaurants()
        {
            using (var container = Bootstrapper.RegisterTypes())
            {
                var service = container.Resolve<IRestaurantService>();

                var results = service.TopThreeRestaurantByAverageReview();

                Approvals.VerifyAll(results, "Restaurants");
            }
        }

        [TestMethod]
        [UseReporter(typeof(DiffReporter))]
        public void AllRestaurants_Returns_AllRestaurantsFromDatabase()
        {
            using (var container = Bootstrapper.RegisterTypes())
            {
                var service = container.Resolve<IRestaurantService>();

                var results = service.AllRestaurants();

                Approvals.VerifyAll(results, "Restaurants");
            }
        }

        [TestMethod]
        [UseReporter(typeof(DiffReporter))]
        public void SearchByString_Returns_AllRestaurantsThatContainStringValue()
        {
            using (var container = Bootstrapper.RegisterTypes())
            {
                var service = container.Resolve<IRestaurantService>();

                var results = service.SearchAll("Jordan");

                Approvals.VerifyAll(results, "Restaurants");
            }
        }

        [TestMethod]
        public void AddRestaurant_Correctly_AddsRestuarantToDatabase()
        {
            var service = new RestaurantService(_mock.Object);

            var resturant = new Restaurant();

            service.AddRestaurant(resturant);

            _mock.Verify(x => x.Add(It.IsAny<Restaurant>()), Times.AtLeastOnce);
        }

        [TestMethod]
        [UseReporter(typeof(DiffReporter))]
        public void SearchByName_GivenString_ReturnsCorrectRestaurant()
        {
            using (var container = Bootstrapper.RegisterTypes())
            {
                var service = container.Resolve<IRestaurantService>();

                var result = service.SearchByName("Billy Bobs Texas BBQ");

                Approvals.Verify(result);
            }
        }
    }
}
