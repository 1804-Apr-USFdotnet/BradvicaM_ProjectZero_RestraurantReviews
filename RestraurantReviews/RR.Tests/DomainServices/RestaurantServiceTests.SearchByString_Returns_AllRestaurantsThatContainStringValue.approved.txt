Restaurants[0] = 
Id: e29b55e4-b684-a698-3834-537d7ecb8423
Name: Jordan Higgins 
Street: 851-2874 Semper Rd.
City: Whangarei
State: NI
ZipCode: 11410
PhoneNumber: 1-434-725-6173
Website: Lee
Rating: 10

